File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_094726/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75

def run_tests():
    # Define test cases: (price, quantity, expected_result, description)
    test_cases = [
        (10.0, 9, 90.0, "Boundary case: quantity 9 (no discount)"),
        (10.0, 10, 90.0, "Boundary case: quantity 10 (10% discount)"),
        (10.0, 49, 441.0, "Boundary case: quantity 49 (10% discount)"),
        (10.0, 50, 375.0, "Boundary case: quantity 50 (25% discount)"),
        (100.0, 5, 500.0, "Standard case: small quantity"),
        (100.0, 100, 7500.0, "Standard case: large quantity"),
    ]

    for price, quantity, expected, description in test_cases:
        actual = calculate_discount(price, quantity)
        
        # Using round to avoid floating point precision issues during comparison
        if round(actual, 2) == round(expected, 2):
            print(f"PASS: {description}")
        else:
            print(f"FAIL: {description} (expected: {expected}, got: {actual})")

if __name__ == "__main__":
    run_tests()

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_094726/typescript/005_unit_test_writer/solution.ts
/**
 * Calculates a bulk discount based on the quantity of items.
 * - Less than 10 items: No discount.
 * - 10 to 49 items: 10% discount.
 * - 50 or more items: 25% discount.
 */
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

function runTests() {
    const tests: TestCase[] = [
        {
            description: "Boundary condition: quantity 9 (no discount)",
            price: 100,
            quantity: 9,
            expected: 900,
        },
        {
            description: "Boundary condition: quantity 10 (10% discount)",
            price: 100,
            quantity: 10,
            expected: 900, // 100 * 10 * 0.9
        },
        {
            description: "Boundary condition: quantity 49 (10% discount)",
            price: 100,
            quantity: 49,
            expected: 4410, // 100 * 49 * 0.9
        },
        {
            description: "Boundary condition: quantity 50 (25% discount)",
            price: 100,
            quantity: 50,
            expected: 3750, // 100 * 50 * 0.75
        },
        {
            description: "Standard case: small quantity",
            price: 10,
            quantity: 2,
            expected: 20,
        },
        {
            description: "Standard case: large quantity",
            price: 10,
            quantity: 100,
            expected: 750, // 10 * 100 * 0.75
        },
    ];

    tests.forEach((test) => {
        const actual = calculateDiscount(test.price, test.quantity);
        
        // We use a small epsilon for floating point comparisons to avoid precision issues
        if (Math.abs(actual - test.expected) < 0.000001) {
            console.log(`PASS: ${test.description}`);
        } else {
            console.log(`FAIL: ${test.description} (expected: ${test.expected}, got: ${actual})`);
        }
    });
}

// Run the test suite
runTests();